fido/fido.py

```diff
@@ -284,14 +295,14 @@ def fetch(
 
     """
 
-    url = _url_to_utf8(url)
+    url = encode_to_bytes(url)
+    method = encode_to_bytes(method)
 
     # Make a copy to avoid mutating the original value
     headers = dict(headers or {})
 
-    # Add basic header values if absent
-    if 'User-Agent' not in headers:
-        headers['User-Agent'] = [user_agent]
+    if 'User-Agent' not in headers and 'user-agent' not in headers:
```

Comment:
Headers do not really care about capitalization
`any(header.lower() == 'user-agent' for header in headers)` ?